Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8828,6 +8828,15 @@ void WebPageProxy::loadAndDecodeImage(WebCore::ResourceRequest&& request, std::o
     });
 }
 
+void WebPageProxy::getInformationFromImageData(Vector<uint8_t>&& data, CompletionHandler<void(String, const Vector<WebCore::IntSize>&)>&& completionHandler)
+{
+    if (!hasRunningProcess())
+        launchProcess(Site(aboutBlankURL()), ProcessLaunchReason::InitialProcess);
+    sendWithAsyncReply(Messages::WebPage::GetInformationFromImageData(WTFMove(data)), [preventProcessShutdownScope = protectedLegacyMainFrameProcess()->shutdownPreventingScope(), completionHandler = WTFMove(completionHandler)] (String typeIdentifier, auto availableSizes) mutable {
```

Comment:
Is the API client expected to be careful to use a separate WKWebView for these queries? If they use an existing WKWebView, the image load will happen in the existing process, which could lead to some security challenges.